proto/connector/v1/connector.proto

```diff
@@ -8,220 +8,236 @@ import "google/protobuf/struct.proto";
+  rpc Stop(Destination.Stop.Request) returns (Destination.Stop.Response);
+  // Teardown signals to the plugin that there will be no more calls to any
+  // other function. After Teardown returns, the plugin should be ready for a
+  // graceful shutdown.
+  rpc Teardown(Destination.Teardown.Request) returns (Destination.Teardown.Response);
 }
 
 // SpecifierPlugin is responsible for returning the plugin specification.
 service SpecifierPlugin {
-    // Specify should return the plugin specification.
-    rpc Specify(Specifier.Specify.Request) returns (Specifier.Specify.Response);
+  // Specify should return the plugin specification.
+  rpc Specify(Specifier.Specify.Request) returns (Specifier.Specify.Response);
 }
 
 message Source {
-    message Configure {
-        message Request {
-            // Config contains the raw plugin settings.
-            map<string, string> config = 1;
-        }
-        message Response {}
+  message Configure {
+    message Request {
+      // Config contains the raw plugin settings.
+      map<string, string> config = 1;
     }
-
-    message Start {
-        message Request {
-            // This is the position of the last record that was successfully
-            // processed. The Source should start producing records after this
-            // position.
-            bytes position = 1;
-        }
-        message Response {}
+    message Response {}
+  }
+
+  message Start {
+    message Request {
+      // This is the position of the last record that was successfully
+      // processed. The Source should start producing records after this
+      // position.
+      bytes position = 1;
     }
-
-    message Run {
-        message Request {
-            // This is the position of the record that was successfully
-            // processed.
-            bytes ack_position = 1;
-        }
-        message Response {
-            // Record contains the record read by the source from the 3rd party
-            // resource.
-            Record record = 1;
-        }
+    message Response {}
+  }
+
+  message Run {
+    message Request {
+      // This is the position of the record that was successfully
+      // processed.
+      bytes ack_position = 1;
     }
-
-    message Stop {
-        message Request {}
-        message Response {}
+    message Response {
+      // Record contains the record read by the source from the 3rd party
+      // resource.
+      Record record = 1;
     }
-
-    message Teardown {
-        message Request {}
-        message Response {}
+  }
+
+  message Stop {
+    message Request {}
+    message Response {
+      // This is the position of the last record in the stream, Conduit
+      // won't process records after this position anymore. After the
+      // record with this position is received by Conduit and all
+      // outstanding acknowledgments were delivered to the connector, the
+      // stream will be closed.
+      bytes last_position = 1;
```

Comment:
This field was added, the stop response contains the position of the last produced record.